econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -0,0 +1,1044 @@
+        est = make_pipeline(StandardScaler(), LogisticRegression(C=model.steps[1][1].C_[0]))
+    if make_regressor:
+        return _RegressionWrapper(est)
+    else:
+        return est
+
+
+def _final_stage():
+    return GridSearchCVList([WeightedLasso(),
+                             RandomForestRegressor(n_estimators=100, random_state=123)],
+                            param_grid_list=[{'alpha': [.001, .01, .1, 1, 10]},
+                                             {'max_depth': [3, 5],
+                                              'min_samples_leaf': [10, 50]}],
+                            cv=5,
+                            scoring='neg_mean_squared_error')
+
+
+# simplification of sklearn's ColumnTransformer that encodes categoricals and passes through selected other columns
+# but also supports get_feature_names with expected signature
+
+
+class _ColumnTransformer(TransformerMixin):
+    def __init__(self, categorical, passthrough):
+        self.categorical = categorical
+        self.passthrough = passthrough
+
+    def fit(self, X):
+        cat_cols = _safe_indexing(X, self.categorical, axis=1)
+        if cat_cols.shape[1] > 0:
+            self.has_cats = True
+            self.one_hot_encoder = OneHotEncoder(
+                drop='first', sparse=False).fit(cat_cols)
+        else:
+            self.has_cats = False
+        self.d_x = X.shape[1]
+        return self
+
+    def transform(self, X):
+        rest = _safe_indexing(X, self.passthrough, axis=1)
+        if self.has_cats:
+            cats = self.one_hot_encoder.transform(
+                _safe_indexing(X, self.categorical, axis=1))
+            return np.hstack((cats, rest))
+        else:
+            return rest
+
+    def get_feature_names(self, names=None):
+        if names is None:
+            names = [f"x{i}" for i in range(self.d_x)]
+        rest = _safe_indexing(names, self.passthrough, axis=0)
+        if self.has_cats:
+            cats = self.one_hot_encoder.get_feature_names(
+                _safe_indexing(names, self.categorical, axis=0))
+            return np.concatenate((cats, rest))
+        else:
+            return rest
+
+
+class CausalAnalysis:
+    """
+    Note: this class is experimental and the API may evolve over our next few releases.
+
+    Gets causal importance of features.
+
+    Parameters
+    ----------
+    feature_inds: array-like of int, str, or bool
+        The features for which to estimate causal effects, expressed as either column indices,
+        column names, or boolean flags indicating which columns to pick
+    categorical: array-like of int, str, or bool
+        The features which are categorical in nature, expressed as either column indices,
+        column names, or boolean flags indicating which columns to pick
+    heterogeneity_inds: array-like of int, str, or bool, or None or list of array-like elements or None, default None
```

Comment:
To avoid confusion I think we should make the default of heterogneity inds be 'all', which means use all variables, and None or empty list, should mean use no variables for heterogeneity.

Reply:
if it's an empty list, forest option will not work? it requires X? and also all the local view will not work neither

Reply:
Sounds good. Let's just elaborate a bit more in the docstring that to use no features for heterogneiety use the empty list.